--- /usr/share/sympa/lib/Upgrade.pm.orig	2011-01-24 09:10:00.000000000 +0100
+++ /usr/share/sympa/lib/Upgrade.pm	2011-01-24 09:10:27.000000000 +0100
@@ -1155,7 +1155,7 @@
     
     my $found_tables = 0;
     foreach $table('user_table', 'subscriber_table', 'admin_table') {
-	if ($checked{$table} || $checked{'public.' . $table}) {
+	if ($checked{$table} || $checked{'public.' . $table} || $checked{$Conf::Conf{'db_user'} . '.' . $table}) {
 	    $found_tables++;
 	}else {
 	    &do_log('err', 'Table %s not found in database %s', $table, $Conf::Conf{'db_name'});
