--- /usr/share/sympa/lib/Sympa/Config/Schema.pm.orig	2023-07-14 15:11:55.779832874 +0000
+++ /usr/share/sympa/lib/Sympa/Config/Schema.pm	2023-07-14 15:12:09.976013042 +0000
@@ -5632,7 +5632,7 @@
         context    => [qw(site)],        # per-robot config is impossible.
         order      => 190.32,
         group      => 'www_other',
-        default    => 'insensitive',
+        default    => 'sensitive',
         gettext_id => 'Password case',
         file       => 'wwsympa.conf',
         gettext_comment =>
