--- /usr/share/sympa/lib/confdef.pm.orig	2014-08-18 21:41:37.482151000 -0700
+++ /usr/share/sympa/lib/confdef.pm	2014-08-18 21:41:46.454195490 -0700
@@ -1498,7 +1498,7 @@
     ##{ one_time_ticket_lockout: not yet implemented. },
     {
         'name'   => 'password_case',
-        'default' => 'insensitive',
+        'default' => 'sensitive',
         'query'  => 'Password case (insensitive | sensitive)',
         'file'   => 'wwsympa.conf',
         'advice' => 'Should not be changed ! May invalid all user password',
