--- /usr/share/sympa/lib/Sympa/ConfDef.pm.orig	2017-12-22 12:43:00.000000000 +0000
+++ /usr/share/sympa/lib/Sympa/ConfDef.pm	2018-11-09 03:23:16.091837833 +0000
@@ -1618,7 +1618,7 @@
         'file'    => 'sympa.conf',
     },
     {   'name'       => 'password_case',
-        'default'    => 'insensitive',
+        'default'    => 'sensitive',
         'gettext_id' => 'Password case',
         'file'       => 'wwsympa.conf',
         #vhost      => '1', # per-robot config is impossible.
