--- /usr/share/sympa/lib/confdef.pm.orig	2014-08-18 21:55:45.902358089 -0700
+++ /usr/share/sympa/lib/confdef.pm	2014-08-18 21:56:47.970665869 -0700
@@ -485,6 +485,10 @@
 	'default' => 'off',
     },
     {
+        'name'    => 'personalize_recipient',
+        'default' => 'off',
+    },
+    {
         'name'    => 'automatic_list_removal',
         'default' => 'none', ## Can be 'if_empty'
         'vhost'   => '1',
--- /usr/share/sympa/lib/List.pm.orig	(revision 10550)
+++ /usr/share/sympa/lib/List.pm	(working copy)
@@ -270,7 +270,7 @@
 		      send editor editor_include delivery_time account topics 
 		      host lang web_archive archive digest digest_max_size available_user_options 
 		      default_user_options msg_topic msg_topic_keywords_apply_on msg_topic_tagging reply_to_header reply_to forced_reply_to * 
-		      verp_rate welcome_return_path remind_return_path merge_feature user_data_source include_file include_remote_file 
+		      verp_rate welcome_return_path remind_return_path merge_feature personalize_recipient user_data_source include_file include_remote_file 
 		      include_list include_remote_sympa_list include_ldap_query
                       include_ldap_2level_query include_sql_query include_admin ttl distribution_ttl creation update 
 		      status serial custom_attribute);
@@ -1416,6 +1416,12 @@
 			       'gettext_id' => "Allow message personnalization",
 			       'group' => 'sending'
 			       },
+	    'personalize_recipient' => {'format' => ['on', 'off'],
+			       'occurence' => '0-1',
+			       'default' => {'conf' => 'personalize_recipient'},
+			       'gettext_id' => "Personalize recipient",
+			       'group' => 'sending'
+			       },
 	    'visibility' => {'scenario' => 'visibility',
 			     'synonym' => {'public' => 'noconceal',
 					   'private' => 'conceal'},
--- /usr/share/sympa/lib/Bulk.pm.orig	2014-08-18 21:59:09.163366005 -0700
+++ /usr/share/sympa/lib/Bulk.pm	2014-08-18 22:24:12.802822151 -0700
@@ -296,6 +296,26 @@
 	return $entity;
     }
 
+    my $list = new List($bulk->{'listname'}, $bulk->{'robot'});
+    my $personalize_recipient = ($list->{'admin'}{'personalize_recipient'} eq 'on');
+
+    if ($entity->head && $entity->head->get('To') && $personalize_recipient) {
+      my $user_details;
+      $user_details->{'email'} = $rcpt;
+      $user_details->{'name'} = $bulk->{'listname'};
+      $user_details->{'domain'} = $bulk->{'robot'};
+
+      # get_subscriber_no_object() return the user's details with the custom attributes
+      my $user = &List::get_subscriber_no_object($user_details);
+
+      if ($user->{'gecos'} && $user->{'email'}) {
+        $entity->head->set('To', $user->{'gecos'} . ' <' . $user->{'email'} . '>');
+      }
+      elsif ($user->{'email'}) {
+        $entity->head->set('To', $user->{'email'});
+      }
+    }
+
     if ($entity->parts) {
 	foreach my $part ($entity->parts) {
 	    unless (_merge_msg($part, $rcpt, $bulk, $data)) {
