--- /usr/share/sympa/default/mhonarc_rc.tt2.orig	2023-07-14 15:19:36.244099577 +0000
+++ /usr/share/sympa/default/mhonarc_rc.tt2	2023-07-14 15:21:03.116644748 +0000
@@ -122,7 +122,7 @@
 <DECODEHEADS>

 <MIMEARGS>
-text/plain; asis=us-ascii nonfixed quote htmlcheck maxwidth=78
+text/plain; asis=us-ascii nonfixed disableflowed quote htmlcheck maxwidth=78
 text/html; asis=us-ascii allownoncidurls
 m2h_external::filter; subdir
 </MIMEARGS>
