--- /usr/share/sympa/lib/Sympa/DatabaseDescription.pm.orig	2018-11-09 06:31:14.349264990 +0000
+++ /usr/share/sympa/lib/Sympa/DatabaseDescription.pm	2018-11-09 06:30:11.367246795 +0000
@@ -147,7 +147,7 @@
                 'order' => 18,
             },
             'include_sources_subscriber' => {
-                'struct' => 'varchar(50)',
+                'struct' => 'varchar(255)',
                 'doc' =>
                     'comma separated list of datasource that contain this subscriber',
                 'order' => 19,
